src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -174,12 +174,12 @@ private void SetGlobalErrorIfTrue(bool arg)
             if (filterOpt == null)
             {
                 WarnUnusedFields(compilation, diagnostics, cancellationToken);
-            }
 
-            MethodSymbol entryPoint = GetEntryPoint(compilation, moduleBeingBuiltOpt, hasDeclarationErrors, diagnostics, cancellationToken);
-            if (moduleBeingBuiltOpt != null && entryPoint != null && compilation.Options.OutputKind.IsApplication())
-            {
-                moduleBeingBuiltOpt.SetPEEntryPoint(entryPoint, diagnostics);
+                MethodSymbol entryPoint = GetEntryPoint(compilation, moduleBeingBuiltOpt, hasDeclarationErrors, diagnostics, cancellationToken);
```

Comment:
Presumably the last 4 only apply if the Main method is specified, so we're losing out on the first 5 if Full Solution Diagnostics are disabled.  I think that's acceptable.